### PR TITLE
Setup international language

### PR DESCRIPTION
**Pull request by ranju30**
Page in french is visible in french.
Need to add french where data is coming from .js file.

### CODE REVIEW

**Comment by gsluthra** [comment]
In this pull request Ranju has ensured that all places where strings were hardcoded in English, have been externalized to a properties file. Of course this means that French file now needs some more translations to be accomplished. 

The only few places where English strings are still hardcoded are Table columns being generated from Javascript. Will fix those soon too.

File: openelis/src/MessageResources.properties [hunk 1]
```diff
@@ -826,7 +826,7 @@ login.notice.notification.LNSP_Haiti  = &nbsp;
 login.password                   = Password
 login.password.expired.date      = Password Expired Date
 login.session.timeout.message    = Your session will expire in:
-login.subTitle                   = Login
+login.subTitle                   = Login As Ranju
```
Comment (gsluthra) [inline openelis/src/MessageResources.properties]: Looks like you committed this by mistake. "Login As Ranju".

**Comment by gsluthra** [comment]
Do we need this file now: "BahmniMessageResources.properties"? Can this not be deleted, and its messages removed to the MessageResources_XX.properties files? Can you please try this out, and let us know.

**Comment by gsluthra** [comment]
Also for table column names, that are hardcoded in Javascript, you can do the following: http://stackoverflow.com/a/23823951/365358

Basically, via JSP, create an HTML element, with custom data attributes, which hold the column translations. Something like:

 `data-column-sample="<bean:message key='dashboard.table.column.sample'/>"`. 

And then in JS read the data-attribute value to get the translated string:

` var translatedCaptionForColumn1 = $("body").attr("data-column-sample");`

**Comment by ranju30** [comment]
Yes, "Login As Ranju" is a mistake, I will change it. I am tring to change
the hardcoded Javascript also.

On Sun, May 22, 2016 at 4:43 PM, Gurpreet <notifications@github.com> wrote:

> Also for table column names, that are hardcoded in Javascript, you can do
> the following: http://stackoverflow.com/a/23823951/365358
>
> Basically, via JSP, create an HTML element, with custom data attributes,
> which hold the column translations. Something like:
>
> data-column-sample="<bean:message key='dashboard.table.column.sample'/>".
>
> And then in JS read the data-attribute value to get the translated string:
>
> var translatedCaptionForColumn1 = $("body").attr("data-column-sample");
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Bahmni/OpenElis/pull/3#issuecomment-220826820>
>



-- 
Regards,

*Ranju Karmakar**STEP Intern.*
Thoughtworks

**Comment by gsluthra** [comment]
It seems by looking at the build.xml, that the BahmniMessageResources.properties is merged with the English MessageResources.properties file, so that one can override existing values with Bahmni specific overrides. (ASKING THE TEAM) Do we need this feature?

**Comment by djazayeri** [comment]
Also, do you want me to put the localizations in transifex, or is what we
are doing with OpenELIS minimal enough not to bother?

-Darius (by phone)

**Comment by gsluthra** [comment]
I think it will be very good for us to perform translations for OpenELIS via transifex,@djazayeri can you help us on this front. There is a single properties file for this. For starters, we need French translations, because French is currently the only non-English language with partial support in OpenELIS. We still need to modify certain parts of the code, to enable reading other languages.

**Comment by ranju30** [comment]
For setting up the default language as 'french' do : 
# UPDATE clinlims.site_information set value='fr-FR' where id=52;

**Comment by djazayeri** [comment]
Okay, just to clarify, are you saying I should add BahmniMessageResources.properties to transifex, but not MessageResources.properties?

**Comment by gsluthra** [comment]
No. The other way round. We want to add MessageResources.properties (and
its current French version too?) to transifex.

Regards
Gurpreet

Community Lead, Bahmni
About me: http://life-lessons.in

**Comment by djazayeri** [comment]
I loaded them at https://www.transifex.com/openmrs/bahmni/openelis/
It showed an error message while uploading the French, but it actually seems to have loaded completely. So, let me know if you see anything problematic there.

However note that in this file there are some bad characters e.g. in "errorpage.previous": https://github.com/Bahmni/OpenElis/blob/master/openelis/src/MessageResources_fr.properties

and these same bad characters show up in Transifex.

(I have not configured the OpenELIS codebase to be able to pull down these translations. Someone needs to do that. It should look something like https://github.com/Bahmni/openmrs-module-bahmniapps/blob/master/.tx/config but more details are here: http://docs.transifex.com/client/config/

**Comment by bharatak** [comment]
We can look merge it in 0.83.. We froze code for this release.

**Comment by gsluthra** [comment]
Added a card in mingle for this: https://bahmni.mingle.thoughtworks.com/projects/bahmni_emr/cards/1918

**Pull request merged**